modules/tinymce/src/core/test/ts/browser/api/dom/RangeExpandTest.ts

```diff
@@ -0,0 +1,91 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Unicode } from '@ephox/katamari';
+import { TinyHooks, TinySelections } from '@ephox/mcagar';
+import { assert } from 'chai';
+
+import RangeUtils from 'tinymce/core/api/dom/RangeUtils';
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.core.api.dom.RangeExpandTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    indent: false,
+    base_url: '/project/tinymce/js/tinymce'
+  }, []);
+
+  const textEntry = 'A B ' + Unicode.nbsp + 'C D';
+
+  const makeBaseSelection = (editor: Editor, startOffset: number, endOffset: number, nodeOffset: number = 0) => {
+    const body = editor.getBody();
+    const range = editor.dom.createRng();
+    const node = body.childNodes[0].childNodes[nodeOffset];
+    range.setStart(node, startOffset);
+    range.setEnd(node, endOffset);
+    return range;
+  };
+
+  const compareRanges = (range1: Range, range2: Range) => {
```

Comment:
I wonder if we could make the output of this assert a bit easier to read. Maybe that should even be part of agar. Like a special assert for ranges. If it has a root and two ranges it could produce paths instead of comparing the ranges directly that way you get a nicer diff displayed together with a HTML dump of the root maybe. We have it for selections in TinyAssertions but not for general ranges. I guess one could inline it here for now and maybe thing about making it generic later on since it seems a bit out of scope for this PR.

Reply:
Do you want to make an issue about it?

Reply:
yes make this a TechDebt task please

Reply:
I can make a ticket about improving range assertions. I think we have similar code to this in a few places.

Reply:
I'll leave it to you then, Spocke.

Reply:
https://ephocks.atlassian.net/browse/TINY-9031